core/state/state_object.go

```diff
@@ -171,9 +219,8 @@ func (c *StateObject) ReturnGas(gas, price *big.Int) {}
 
 func (self *StateObject) Copy(db trie.Database) *StateObject {
 	stateObject := NewStateObject(self.address, db)
-	stateObject.balance.Set(self.balance)
-	stateObject.codeHash = common.CopyBytes(self.codeHash)
-	stateObject.nonce = self.nonce
+	stateObject.data = self.data
+	stateObject.data.Balance.Set(self.data.Balance)
```

Comment:
Memo to myself (looking at commits separately), the above assignment already shallow copied everything, so this balance set shouldn't actually do anything. Will trace in newer commits since this changes later.